src/components/CreateForm.tsx

```diff
@@ -56,39 +56,58 @@ class CreateForm extends React.Component<CreateScheduleProps, CreateScheduleStat
                         history.push('/vote?key=' + key + '&b=' + r.receipt.blockNumber)
                     })
             })
+            
+            // `.then()` w js działa zarówno jak map, jak i flatMap, zatem mozna to zrobic nieco ładniej (specjalnie zakomentowane, zeby latwiej bylo Ci sie polapac, co dodałem):
+            // this.props.contract
+            //     .deployed()
+            //     .then(instance => instance.create(ballotId, this.state.name, this.state.options, { from: this.props.account, gas: 1334400 }))
+            //     .then(r => {
+            //         console.log("Contract executed")
+            //         history.push('/vote?key=' + key + '&b=' + r.receipt.blockNumber)
+            //     });
         }
     }
 
     addNewDate() {
         if (this.validInput()) {
+            // ŹLE! w linii 76 modyfikujesz stan bezposrednio, a tak nie wolno!
+            // przeciez newOptions wskazuje na this.state.options :)
             let newOptions: Array<string> = this.state.options
             newOptions.push(this.state.newOption)
             this.setState({ options: newOptions, newOption: this.now() })
+
+            // zrób to tak (odkomentuj):
+            // this.setState({
+            //     options: [...this.state.options, this.state.newOption], // spread operator (...this.state.options) robi shallow copy
+            //     newOption: this.now()
+            // });
         }
     }
 
     validInput(): boolean {
         if (this.state.options.some(x => x === this.state.newOption)) {
             this.setState({ formError: true, formErrorMsg: "Duplicates not allowed" })
             return false
-        }
+        } // else od tej samej linii
         else {
             this.setState({ formError: false })
             return true
         }
     }
 
-    handleInputChange(event: any) {
+    handleInputChange(event: any) { // ten typ 'any' troche wadzi, sprobowalbym z `React.SyntheticEvent<HTMLInputElement>` oraz wtedy `event.currentTarget.value` (zobacz https://stackoverflow.com/questions/42081549/typescript-react-event-types#answer-42085792)
         this.setState({ newOption: event.target.value })
     }
 
-    handleNameChange(event: any) {
+    handleNameChange(event: any) { // jw
         this.setState({ name: event.target.value })
```

Comment:
Pytanie: To mi wygląda na pattern: mamy sobie jakiś input, którego onChange zmienia pole w state. Dałoby się to jakoś zgeneralizować? Np. robiąc na inpucie coś w stylu na inpucie `onChange = {updateStateField(name)}`